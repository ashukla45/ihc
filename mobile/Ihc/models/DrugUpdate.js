import {stringDate} from '../util/Date';
export default class DrugUpdate {
  // Insert any class methods here

  static extractFromForm(form, patientKey) {
    const update = Object.assign({}, form);
    update.patientKey = patientKey;
    update.date = stringDate(new Date());
    return update;
  }

  static getInstance() {
    return {
      patientKey: 'firstname&father&mother&20000101',
      name: 'tylenol',
      date: '20180101',
      dose: '1',
      frequency: '1',
      duration: '1',
      lastUpdated: new Date().getTime()
    };
  }
}

DrugUpdate.schema = {
  name: 'DrugUpdate',
  properties: {
    patientKey: 'string',
    name: 'string', // drug name
    date: 'string',
    dose: 'string',
    frequency: 'string',
    duration: 'string',
    notes: 'string?',
    lastUpdated: 'int',
  }
};
