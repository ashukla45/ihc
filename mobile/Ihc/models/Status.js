import {stringDate} from '../util/Date';
import Patient from './Patient';
/*
 * A patient's status as they go through their appointment
 */
export default class Status {
  // Insert any class methods here

  static newStatus(patient) {
    const obj = {
      patientKey: patient.key,
      name: Patient.fullName(patient),
      birthday: patient.birthday,
      date: stringDate(new Date()),
      active: true,
      checkinTime: new Date().getTime(),
      lastUpdated: new Date().getTime(),
    };
    return obj;
  }
}

Status.schema = {
  name: 'Status',
  properties: {
    patientKey: 'string',
    name: 'string',
    birthday: 'string', // For convenience in patient select table
    date: 'string',
    active: 'bool',
    checkinTime: 'int', // Timestamp, milliseconds
    triageCompleted: 'int?', // timestamp for when completed
    doctorCompleted: 'int?',
    pharmacyCompleted: 'int?',
    notes: 'string?',
    lastUpdated: 'int',
  }
};
