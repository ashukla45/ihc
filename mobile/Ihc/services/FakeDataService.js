/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests
export function getPatient(patientKey) {
  return {};
}

export function getTriage(patientKey, date) {
  return {};
}

export function getSoap(patientKey, date) {
  return {};
}

export function getPatients(param) {
  return []; // Return whatever sample data you want
}

export function getUpdates(param) {
  return []; // Return whatever sample data you want
}

/*
 * active_only is true if we only want the statuses of patients that are
 * currently checked in
 */
export function getPatientSelectRows() {
  const arr =
    [
      newFakePatientSelectRow('Jose'),
      newFakePatientSelectRow('Bob'),
      newFakePatientSelectRow('Matt'),
      newFakePatientSelectRow('Coco'),
      newFakePatientSelectRow('Brandon'),
      newFakePatientSelectRow('Evan'),
      newFakePatientSelectRow('Sarah'),
      newFakePatientSelectRow('Daniel'),
      newFakePatientSelectRow('Neil'),
      newFakePatientSelectRow('William')
    ];

  const columnOrder = ['name', 'birthday', 'checkin_time', 'triage_completed',
    'doctor_completed', 'pharmacy_completed', 'notes'];
  const toReturn = createMatrix(arr, columnOrder);

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, toReturn);
  });
}

export function getMedicationUpdates() {
  const arr = [
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Advil'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Cold meds'),
  ];

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, arr);
  });
}

const medsCount = 0;
function newFakeMedicationUpdate(name) {
  medsCount = ++medsCount % 3;

  return {
    name: name,
    date: `2018020${medsCount}`,
    dose: '10mg',
    frequency: 'bid',
    duration: '1 month',
    notes: 'yes'
  };
}

const numFakePatients = 0;
// Currently ignore triage/doctor/pharmacy params
function newFakePatientSelectRow(name, triage = false, doctor = false,
  pharm = false, notes = '') {
  numFakePatients++;
  return {
    name: name,
    birthday: '20180201',
    checkin_time: new Date().getTime(),
    triage_completed: numFakePatients % 2 === 0 ? new Date().getTime() : null,
    doctor_completed: numFakePatients % 4 === 0 ? new Date().getTime() : null,
    pharmacy_completed: numFakePatients % 8 === 0 ? new Date().getTime() : null,
    notes: notes
  };
}

function createMatrix(data, columnOrder) {
  return data.map((obj) => columnOrder.map( (key) => obj[key] ));
}
/* eslint-enable */
