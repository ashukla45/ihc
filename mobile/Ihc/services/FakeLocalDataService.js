/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests
export function createPatient(patient) {
}

export function signinPatient(patientForm) {
}

export function updateStatus(patientKey, strDate, field, value) {
}

export function createDrugUpdate(update) {
}

export function updateSoap(update) {
}

export function getSoap(patientKey, strDate) {
}

export function updateTriage(update) {
}

export function getTriage(patientKey, strDate) {
}

export function getPatient(patientKey) {
}

export function getPatientsToUpload() {
}

export function lastSynced() {
}

export function handleDownloadedPatients(patients) {
}

/*
 * active_only is true if we only want the statuses of patients that are
 * currently checked in
 */
export function getPatientSelectRows() {
  const arr =
    [
      newFakePatientSelectRow('Jose'),
      newFakePatientSelectRow('Bob'),
      newFakePatientSelectRow('Matt'),
      newFakePatientSelectRow('Coco'),
      newFakePatientSelectRow('Brandon'),
      newFakePatientSelectRow('Evan'),
      newFakePatientSelectRow('Sarah'),
      newFakePatientSelectRow('Daniel'),
      newFakePatientSelectRow('Neil'),
      newFakePatientSelectRow('William')
    ];

  const columnOrder = ['name', 'birthday', 'checkin_time', 'triage_completed',
    'doctor_completed', 'pharmacy_completed', 'notes'];
  const toReturn = createMatrix(arr, columnOrder);

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, toReturn);
  });
}

export function getMedicationUpdates() {
  const arr = [
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Advil'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Cold meds'),
  ];

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, arr);
  });
}

const medsCount = 0;
function newFakeMedicationUpdate(name) {
  medsCount = ++medsCount % 3;

  return {
    name: name,
    date: `2018020${medsCount}`,
    dose: '10mg',
    frequency: 'bid',
    duration: '1 month',
    notes: 'yes'
  };
}

const numFakePatients = 0;
// Currently ignore triage/doctor/pharmacy params
function newFakePatientSelectRow(name, triage = false, doctor = false,
  pharm = false, notes = '') {
  numFakePatients++;
  return {
    name: name,
    birthday: '20180201',
    checkin_time: new Date().getTime(),
    triage_completed: numFakePatients % 2 === 0 ? new Date().getTime() : null,
    doctor_completed: numFakePatients % 4 === 0 ? new Date().getTime() : null,
    pharmacy_completed: numFakePatients % 8 === 0 ? new Date().getTime() : null,
    notes: notes
  };
}

function createMatrix(data, columnOrder) {
  return data.map((obj) => columnOrder.map( (key) => obj[key] ));
}
/* eslint-enable */
