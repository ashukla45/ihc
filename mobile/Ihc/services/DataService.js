/* global */
// Returns either fakeLocalData or RealData depending on config file
// Config file:
// "FakeDataServices": "true" or "false"
import * as fakeLocalData from '../services/FakeLocalDataService';
import * as localDataService from '../services/LocalDataService';

import * as fakeServerData from '../services/FakeServerDataService';
import * as serverDataService from '../services/ServerDataService';

import config from '../config.json';

// Dont use the Fakes in tests, so that we can test the LocalDataService itself,
// but should stub out its methods for the other tests
let localData = (config.fakeLocalDataService === 'true') ? fakeLocalData : localDataService;

// If we are testing the server data service, then use the real one
let serverData = (config.fakeServerDataService === 'true' && config.testingServerDataService == 'false')
  ? fakeServerData : serverDataService;

export {localData, serverData};
