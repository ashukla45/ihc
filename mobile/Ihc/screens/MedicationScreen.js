import React, { Component } from 'react';
import {
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import {localData, serverData} from '../services/DataService';
import MedicationTable  from '../components/MedicationTable';
import Container from '../components/Container';
import {stringDate} from '../util/Date';
import DrugUpdate from '../models/DrugUpdate';

export default class MedicationScreen extends Component<{}> {
  /*
   * Props:
   * name: patient's name for convenience
   * patientKey: string of patient's key
   */
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
      updates: [],
      errorMsg: null,
      successMsg: null,
      medicationCheckmarks: [],
      todayDate: stringDate(new Date())
    };
    this.props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

  // Reload table after new medication updates
  onNavigatorEvent(event) {
    if (event.id === 'willAppear') {
      this.loadMedications();
    }
  }

  refillMedication = (prevDrugUpdate) => {
    const newUpdate = Object.assign({}, prevDrugUpdate);
    newUpdate.date = this.state.todayDate;

    try {
      localData.createDrugUpdate(newUpdate);
      this.loadMedications();
      this.setState({errorMsg: null});
    } catch(e) {
      this.setState({errorMsg: e.message});
    }
  }

  changeMedication = (prevDrugUpdate) => {
    this.props.navigator.push({
      screen: 'Ihc.MedicationUpdateScreen',
      title: 'Medication',
      passProps: {
        drugUpdate: prevDrugUpdate,
        action: 'change',
        patientKey: this.props.patientKey,
        name: this.props.name
      }
    });
  }

  discontinueMedication = (prevDrugUpdate) => {
    // Add a dummy drug update to updates prop
    const newUpdate = DrugUpdate.getDiscontinueDummy(prevDrugUpdate);
    this.setState({
      updates: this.state.updates.concat([newUpdate])
    });
  }

  createNewMedication = () => {
    this.props.navigator.push({
      screen: 'Ihc.MedicationUpdateScreen',
      title: 'Medication',
      passProps: {
        action: 'new',
        patientKey: this.props.patientKey,
        name: this.props.name
      }
    });
  }

  loadMedications = () => {
    this.setState({ loading: true });
    let updates = localData.getMedicationUpdates(this.props.patientKey);
    let statusObj = localData.getStatus(this.props.patientKey, this.state.todayDate);
    const checkmarks = statusObj.medicationCheckmarks;
    this.setState({
      updates: updates,
      loading: false,
      medicationCheckmarks: checkmarks,
    });
  }

  componentDidMount() {
    this.loadMedications();
  }

  saveCheckmarks = () => {
    // Local checkmark saves are handled in MedicationTable directly.
    const statusObj = localData.getStatus(this.props.patientKey, this.state.todayDate);

    this.setState({
      loading: true,
      errorMsg: null,
      successMsg: null,
    });

    serverData.updateStatus(statusObj)
      .then( () => {
        if(this.state.loading) {
          this.setState({
            successMsg: 'Saved',
            errorMsg: null,
            loading: false,
            showRetryButton: false
          });
        }
      })
      .catch( (e) => {
        if(this.state.loading) {
          localData.markPatientNeedToUpload(this.props.patientKey);
          this.setState({
            successMsg: null,
            errorMsg: e.message,
            loading: false,
            showRetryButton: true
          });
        }
      });
  }

  // station: 'Doctor' or 'Pharmacy'
  updateStatus(station) {
    this.setState({
      loading: true,
      errorMsg: null,
      successMsg: null,
    });

    let statusObj = {};
    if(station != 'Doctor' && station != 'Pharmacy') {
      throw new Error(`Received invalid station: ${station}`);
    }

    const fieldName = station === 'Doctor' ? 'doctorCompleted' : 'pharmacyCompleted';
    try {
      statusObj = localData.updateStatus(this.props.patientKey, this.state.todayDate,
        fieldName, new Date().getTime());
    } catch(e) {
      this.setState({errorMsg: e.message, successMsg: null, loading: false});
      return;
    }

    serverData.updateStatus(statusObj)
      .then( () => {
        if(this.state.loading) {
          this.setState({
            successMsg: `${station} marked as completed`,
            errorMsg: null,
            loading: false,
            showRetryButton: false
          });
        }
      })
      .catch( (e) => {
        if(this.state.loading) {
          localData.markPatientNeedToUpload(this.props.patientKey);
          this.setState({
            successMsg: null,
            errorMsg: e.message,
            loading: false,
            showRetryButton: true
          });
        }
      });
  }

  completed = () => {
    this.updateStatus('Doctor');
  }

  filled = () => {
    this.updateStatus('Pharmacy');
  }

  // If Loading was canceled, we want to show a retry button
  setLoading = (val, canceled) => {
    this.setState({loading: val, showRetryButton: canceled});
  }

  setMsg = (type, msg) => {
    const obj = {};
    obj[type] = msg;
    const other = type === 'successMsg' ? 'errorMsg' : 'successMsg';
    obj[other] = null;
    this.setState(obj);
  }

  render() {
    return (
      <Container loading={this.state.loading}
        errorMsg={this.state.errorMsg}
        successMsg={this.state.successMsg}
        setLoading={this.setLoading}
        setMsg={this.setMsg}
        patientKey={this.props.patientKey}
        showRetryButton={this.state.showRetryButton}
      >

        <View style={styles.headerContainer}>
          <Text style={styles.title}>
            {this.props.name}'s Medications
          </Text>

          <Text>R: Refill, D: Change Dose, X: Cancel</Text>
          <Text>T: Taking, N: Not Taking, I: Taking incorrectly</Text>
        </View>

        <ScrollView contentContainerStyle={styles.tableContainer} horizontal>
          <MedicationTable style={styles.table}
            refill={this.refillMedication}
            change={this.changeMedication}
            discontinue={this.discontinueMedication}
            updates={this.state.updates}
            medicationCheckmarks={this.state.medicationCheckmarks}
            patientKey={this.props.patientKey}
          />
        </ScrollView>

        <View style={styles.footerContainer}>
          <TouchableOpacity
            style={styles.buttonContainer}
            onPress={this.saveCheckmarks}>
            <Text style={styles.button}>Save checkmarks</Text>
          </TouchableOpacity>

          <TouchableOpacity
            style={styles.buttonContainer}
            onPress={this.createNewMedication}>
            <Text style={styles.button}>New Medication</Text>
          </TouchableOpacity>

          <TouchableOpacity
            style={styles.buttonContainer}
            onPress={this.completed}>
            <Text style={styles.button}>Completed Prescribing</Text>
          </TouchableOpacity>

          <TouchableOpacity
            style={styles.buttonContainer}
            onPress={this.filled}>
            <Text style={styles.button}>Filled</Text>
          </TouchableOpacity>
        </View>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  headerContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
  },
  tableContainer: {
    flex: 0,
  },
  buttonContainer: {
    width: 120,
    margin: 4,
    padding: 8,
    elevation: 4,
    borderRadius: 2,
    backgroundColor: '#2196F3',
    height: 40,
  },
  button: {
    fontWeight: '500',
    color: '#fefefe',
    textAlign: 'center',
  },
  footerContainer: {
    flex: 1,
    flexDirection: 'row',
    height: 40,
    margin: 4,
  },
});
