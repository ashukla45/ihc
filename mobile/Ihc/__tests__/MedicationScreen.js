import 'react-native';
import React from 'react';
import MedicationScreen from '../screens/MedicationScreen';

import DrugUpdate from '../models/DrugUpdate';
import Status from '../models/Status';
import sinon from 'sinon';
import {localData} from '../services/DataService';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  sinon.useFakeTimers(100);
  const drug1 = DrugUpdate.getInstance();
  const drug2 = DrugUpdate.getInstance();
  sinon.stub(localData, 'getMedicationUpdates').returns([drug1, drug2]);
  sinon.stub(localData, 'getStatus').returns(Status.newStatus({key: 'key'}));

  const fakeNavigator = { setOnNavigatorEvent: (param) => {} };
  const json = renderer.create(
    <MedicationScreen navigator={fakeNavigator} />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
