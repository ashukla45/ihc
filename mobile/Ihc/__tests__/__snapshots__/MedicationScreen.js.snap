// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          's Medications
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          R: Refill, D: Change Dose, X: Cancel
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          T: Taking, N: Not Taking, I: Taking incorrectly
        </Text>
      </View>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flex": 0,
          }
        }
        horizontal={true}
      >
        <View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#adada0",
                  "borderWidth": 1,
                  "flex": 0,
                  "flexDirection": "column",
                  "width": 100,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#dbdbdb",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "maxHeight": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  T/N/I
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#dddddd",
                    "borderWidth": 1,
                    "flex": 0,
                    "flexDirection": "row",
                    "height": 60,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "flex-start",
                        "borderColor": "red",
                        "borderWidth": 1,
                      },
                      undefined,
                    ]
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "flex-start",
                        "borderColor": "red",
                        "borderWidth": 1,
                      },
                      undefined,
                    ]
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "flex-start",
                        "borderColor": "red",
                        "borderWidth": 1,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#adada0",
                  "borderWidth": 1,
                  "flex": 0,
                  "flexDirection": "column",
                  "width": 100,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#dbdbdb",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "maxHeight": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Drug name
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#dddddd",
                    "borderWidth": 1,
                    "flex": 0,
                    "flexDirection": "row",
                    "height": 60,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  tylenol
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#adada0",
                  "borderWidth": 1,
                  "flex": 0,
                  "flexDirection": "column",
                  "width": 100,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#dbdbdb",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "maxHeight": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Actions
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#dddddd",
                    "borderWidth": 1,
                    "flex": 0,
                    "flexDirection": "row",
                    "height": 60,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#2196F3",
                      "borderRadius": 2,
                      "elevation": 4,
                      "flex": 1,
                      "margin": 2,
                      "opacity": 1,
                      "padding": 4,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#fefefe",
                        "fontWeight": "500",
                        "textAlign": "center",
                      }
                    }
                  >
                    R
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#2196F3",
                      "borderRadius": 2,
                      "elevation": 4,
                      "flex": 1,
                      "margin": 2,
                      "opacity": 1,
                      "padding": 4,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#fefefe",
                        "fontWeight": "500",
                        "textAlign": "center",
                      }
                    }
                  >
                    D
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#2196F3",
                      "borderRadius": 2,
                      "elevation": 4,
                      "flex": 1,
                      "margin": 2,
                      "opacity": 1,
                      "padding": 4,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#fefefe",
                        "fontWeight": "500",
                        "textAlign": "center",
                      }
                    }
                  >
                    X
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#adadad",
                  "borderWidth": 1,
                  "flex": 1,
                  "flexDirection": "column",
                  "minWidth": 250,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#dbdbdb",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "maxHeight": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  19691231
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#dddddd",
                    "borderWidth": 1,
                    "flex": 0,
                    "flexDirection": "row",
                    "height": 60,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  No medications for today
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#adadad",
                  "borderWidth": 1,
                  "flex": 1,
                  "flexDirection": "column",
                  "minWidth": 250,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#dbdbdb",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "maxHeight": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  20180101
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#dddddd",
                    "borderWidth": 1,
                    "flex": 0,
                    "flexDirection": "row",
                    "height": 60,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#adadad",
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                      "minWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    1
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#adadad",
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                      "minWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    1
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#adadad",
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                      "minWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    1
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#adadad",
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "minWidth": 150,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
            "margin": 4,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 40,
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fefefe",
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            Save checkmarks
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 40,
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fefefe",
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            New Medication
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 40,
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fefefe",
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            Completed Prescribing
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 40,
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fefefe",
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            Filled
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
